### PR TITLE
MAGECLOUD-3535:  Move Service Version Validation Higher and Merge ES/PHP Validation

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description

Added validation of services version according to version mapping. Moved service configuration and version to the separate classes. 

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3535
### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-127

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Service/RabbitMq.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Service;
+
+use Magento\MagentoCloud\Config\Environment;
+
+/**
+ *
```
Comment (NadiyaS) [inline src/Service/RabbitMq.php:11]: add description

File: src/Service/RabbitMq.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getVersion(): string
+    {
+        if ($this->version === null) {
+            $this->version = '0';
```
Comment (NadiyaS) [inline src/Service/RabbitMq.php]: Why do we return 0 if service is not available?

File: src/Service/Redis.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     * @return array
+     */
+    public function getSlaveConfiguration(): array
+    {
+        return $this->environment->getRelationship(self::RELATIONSHIP_SLAVE_KEY)[0] ?? [];
+    }
+
+    /**
+     * Returns version of the service.
+     *
```
Comment (NadiyaS) [inline src/Service/Redis.php:56]: Add {@inheritdoc} with description of returned value

**Review by NadiyaS — APPROVED** [review]

File: src/Service/ServiceInterface.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    const NAME_NGINX = 'nginx';
+    const NAME_REDIS = 'redis';
+    const NAME_ELASTICSEARCH = 'elasticsearch';
+    const NAME_RABBITMQ = 'rabbitmq';
+    const NAME_NODE = 'node';
+    const NAME_VARNISH = 'varnish';
+
+    /**
+     * Returns service configuration.
+     * Returns an empty array if service isn't configured
```
Comment (BaDos) [inline src/Service/ServiceInterface.php:24]: Please, use dot-pattern everywhere or do not use anywhere.
Make a choice and follow it.

**Review by BaDos — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Comment by shiftedreality** [comment]
Fails with Magento 2.1 reported in https://magento2.atlassian.net/browse/MAGECLOUD-3875

**Pull request merged**